src/Smalot/PdfParser/Page.php

```diff
@@ -216,8 +216,56 @@ public function getText(self $page = null): string
         return '';
     }
 
+    /**
+     * Return True if the current page is a (setasign\Fpdi\Fpdi) FPDI/FPDF document
+     * 
+     * @return bool true is the current page is a FPDI/FPDF document
+     */
+    public function isFpdf(): Bool
+    {
+        if (array_key_exists("Producer", $this->document->getDetails(true)) and 
+            is_string($this->document->getDetails(true)["Producer"]) and 
+            str_starts_with($this->document->getDetails(true)["Producer"], "FPDF")) {
+                return true;
+            }
+        return false;
+    }
+
+    /**
+     * Return the page number of the PDF document of the page object
+     * 
+     * @return int the page number
+    */
+    public function getPageNumber(): int 
+    {
+        $pages = $this->document->getPages();
+        $numOfPages = count($pages);
+        for ($pageNum = 0; $pageNum < $numOfPages; $pageNum++) {
+            if ($pages[$pageNum] === $this) {
+                break;
+            }
+        }
+        return $pageNum;
+    }
```

Comment:
What does it mean if `$pageNum` is `0`?

Reply:
It means that it is the first page.  This is a work around for this bug.  The reason, is because the whole information of the Text Matrix (the content), is not save in the page, but in the xObject array inside the page, but is in array of pages, so you need the page number... but in the Page object, there isn't an attribute that tells you which page of the document the page is.

This is just for the fpdf/fpdi documents.